diff --git a/lib/std/os/linux/bpf.zig b/lib/std/os/linux/bpf.zig
index 4cb9dbf5c..147bf610d 100644
--- a/lib/std/os/linux/bpf.zig
+++ b/lib/std/os/linux/bpf.zig
@@ -458,7 +458,7 @@ pub const Insn = packed struct {
         else
             ImmOrReg{ .imm = src };
 
-        const src_type = switch (imm_or_reg) {
+        const src_type: u8 = switch (imm_or_reg) {
             .imm => K,
             .reg => X,
         };
